### ISSUE TITLE
json: `%` not called in 0.18.0

### ISSUE THREAD

**Issue by brentp** [nim-lang/Nim#7790]
The code below gives the expected result in 0.17.2 (sets keys to "changed-xs" and "changed-ys") but not in 0.18.0 where it is still serialized to "xs" and "ys". The code below uses generics, but it also happens when xs and ys are `seq[int]`.

```Nim
import json
import tables

type
  Data*[T: SomeNumber] = ref object
    xs*: seq[T]
    ys*: seq[T]

proc `%`*(d: Data): JsonNode =
  var fields = initOrderedTable[string, JsonNode](2)
  fields["changed-xs"] = % d.xs
  fields["changed-ys"] = % d.ys
  result = JsonNode(kind:Jobject, fields:  fields)


var d = Data[int](xs:new_seq[int](10), ys: new_seq[int](10))

var j = %*d
echo j
```

**Comment by jangko** [comment]
the problem is someone change the bindSym behavior between 0.17.2 and 0.18.0 without documenting it(or don't aware about the new behavior)
current behavior of bindSym ignore `%` from caller scope and only bind symbols from the json module scope.
default bindSym rule is brClosed (only the symbols in current scope are bound). 
but that is also not clear what is the meaning of `current scope`, the `scope of module defining the macro using bindSym` or the `scope of the macro caller`?

**Comment by brentp** [comment]
I think this is closed by: #7965 , correct?

**Comment by Yardanico** [comment]
Yes, correct! Thanks for noticing.
